general/ai/cli.go

```diff
@@ -59,25 +64,75 @@ func HowCmd(c *cli.Context) error {
 			return err
 		}
 		log.Output(coreutils.PrintLink(llmAnswer))
+
+		err = sendFeedback()
+		if err != nil {
+			return err
+		}
+
 		log.Output("\n" + coreutils.PrintComment("-------------------") + "\n")
 	}
 }
 
+type QuestionBody struct {
+	Question string `json:"question"`
+}
+
 func askQuestion(question string) (response string, err error) {
-	contentBytes, err := json.Marshal(QuestionBody{Question: question})
+	return sendRestAPI(ask, QuestionBody{Question: question})
+}
+
+type FeedbackBody struct {
+	IsGoodResponse bool `json:"is_good_response"`
+}
+
+func sendFeedback() (err error) {
+	isGoodResponse, err := getUserFeedback()
+	if err != nil {
+		return err
+	}
+	_, err = sendRestAPI(feedback, FeedbackBody{IsGoodResponse: isGoodResponse})
+	return
+}
+
+func getUserFeedback() (bool, error) {
+	// Customize the template to place the options on the same line as the question
+	templates := &promptui.SelectTemplates{
+		Label:    "{{ . }}",           // Question label
+		Active:   "👉 {{ . | cyan  }}", // Highlight the active item
+		Inactive: "   {{ . }}",        // No special formatting for inactive items
+	}
+
+	prompt := promptui.Select{
+		Label:     "Rate this response:",                                // The question with emojis on the same line
+		Items:     []string{"👍 Good response!", "👎 Could be better..."}, // Items to select
+		Templates: templates,                                            // Apply the customized template
+		HideHelp:  true,                                                 // Hide the help message
+	}
+	selected, _, err := prompt.Run()
+	if err != nil {
+		return false, err
+	}
+	return selected == 0, nil
+}
+
+func sendRestAPI(apiType ApiType, content interface{}) (response string, err error) {
+	contentBytes, err := json.Marshal(content)
 	if errorutils.CheckError(err) != nil {
 		return
 	}
 	client, err := httpclient.ClientBuilder().Build()
 	if errorutils.CheckError(err) != nil {
 		return
 	}
-	req, err := http.NewRequest(http.MethodPost, cliAiAskApiPath, bytes.NewBuffer(contentBytes))
+	req, err := http.NewRequest(http.MethodPost, path.Join(cliAiAskApiPath, string(apiType)), bytes.NewBuffer(contentBytes))
```

Comment:
Let's use jferog-client-go's [SendPost](https://github.com/jfrog/jfrog-client-go/blob/b1cffdca17f4959c1ff97c32ac0cd06b602d539c/http/jfroghttpclient/client.go#L34) API instead of rewriting it. This Go client has gone though multiple fixes and improvement over the year.

Reply:
i cant use it, its not a generic HTTP client its a JFrog Platform client. It has runPreRequestInterceptors and retry mechanism and special jfrog auth methods and headers
@eyalbe4

Reply:
👍